net_printer/README.md

```diff
@@ -0,0 +1,116 @@
+         }] belPin  SLICE_X90Y91 EFF D
+============================================================
+```
+
+In this example `np.py` is invoked on the netlist `vtr_mcml_rwroute.phys` (This
+netlist that can be generated from a global `make`,
+see [the contest website](https://xilinx.github.io/fpga24_routing_contest/index.html))
+in order to printout the net named `u_calc/boundaryChecker/r_ux__57_reg[21]_srl27___u_calc_dropSpin_photon29_o_sleftz_reg_r_n_0`.
+Since this is a fully routed net the output shows one source (`Source: 0`) and
+no stubs. Following the source is a list of all of the edges (FPGAIF
+routeSegements) in the route tree. The symbol `[{` indicates root of the tree.
+The symbol `{` indicates the begining of a branch and the symbol `}` indicates
+the end of a branch. Finally, the symbol `}]` indicates the end of the tree.
+
+In the example above we see that the first edge in the tree is:
+```
+belPin  SLICE_X84Y94 C6LUT O6
+```
+which is immediately followed by a branch containing the edges:
+```
+sitePIP SLICE_X84Y94 OUTMUXC D6 False
+belPin  SLICE_X84Y94 OUTMUXC OUT
+sitePin SLICE_X84Y94 CMUX
+```
+And then a branch with a single edge:
+```
+belPin  SLICE_X84Y94 OUTMUXC D6
+```
+Before we return to the main branch:
+```
+sitePin SLICE_X84Y94 C_O
+```
+
+Thus, if you were to draw the first part of the above tree it would look like:
+```
+(belPin  SLICE_X84Y94 C6LUT O6)------------+-----------------------------------------+
+           |                               |                                         |
+           V                               V                                         V
+(sitePin SLICE_X84Y94 C_O)      (sitePIP SLICE_X84Y94 OUTMUXC D6 False)   (belPin  SLICE_X84Y94 OUTMUXC D6)
+           |                               |
+	   V                               V
+          ...                   (belPin  SLICE_X84Y94 OUTMUXC OUT)
+                                           |
+					   V
+                                (sitePin SLICE_X84Y94 CMUX)
+```
+
+`np.py` can also print out unrouted nets, for example:
+```
+$ python3 np.py ../vtr_mcml_unrouted.phys u_calc/boundaryChecker/r_ux__57_reg[21]_srl27___u_calc_dropSpin_photon29_o_sleftz_reg_r_n_0
+============================================================
+Route tree for net: u_calc/boundaryChecker/r_ux__57_reg[21]_srl27___u_calc_dropSpin_photon29_o_sleftz_reg_r_n_0
+
+    Source: 0
+    [{      belPin  SLICE_X84Y94 C6LUT O6
+     {      sitePIP SLICE_X84Y94 OUTMUXC D6 False
+         }  belPin  SLICE_X84Y94 OUTMUXC OUT
+     {   }  belPin  SLICE_X84Y94 OUTMUXC D6
+            belPin  SLICE_X84Y94 C_O C_O
+         }] sitePin SLICE_X84Y94 C_O
+
+    Stub: 0
+    [{      sitePin SLICE_X90Y91 EX
+            belPin  SLICE_X90Y91 EX EX
+     {   }  belPin  SLICE_X90Y91 FFMUXE1 BYP
+            sitePIP SLICE_X90Y91 FFMUXE1 BYP False
+            belPin  SLICE_X90Y91 FFMUXE1 OUT1
+         }] belPin  SLICE_X90Y91 EFF D
```

Comment:
For me:
```
    Source: 0
    [{      belPin  SLICE_X84Y94 C6LUT O6
     {      belPin  SLICE_X84Y94 OUTMUXC D6
            sitePIP SLICE_X84Y94 OUTMUXC D6 False
         }  belPin  SLICE_X84Y94 OUTMUXC OUT
            belPin  SLICE_X84Y94 C_O C_O
         }] sitePin SLICE_X84Y94 C_O

    Stub: 0
    [{      sitePin SLICE_X90Y91 EX
            belPin  SLICE_X90Y91 EX EX
            belPin  SLICE_X90Y91 FFMUXE1 BYP
            sitePIP SLICE_X90Y91 FFMUXE1 BYP False
            belPin  SLICE_X90Y91 FFMUXE1 OUT1
         }] belPin  SLICE_X90Y91 EFF D
```

Reply:
please see: 21375b89bb708c37ea993de441617d81958e7896